src/test/java/org/fogbowcloud/manager/core/processors/FulfilledProcessorTest.java

```diff
@@ -99,355 +95,365 @@ public void tearDown() {
-        
+
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
 
-        /** here may be a false positive depending on how long the machine will take to run the test */
-        Thread.sleep(MAX_SLEEP_TIME);
-
-        Assert.assertNull(this.fulfilledOrderList.getNext());
+        /**
+         * here may be a false positive depending on how long the machine will take to run the test
+         */
+        this.thread.sleep(MAX_SLEEP_TIME);
 
+        // verify
         Order test = this.failedOrderList.getNext();
         Assert.assertNotNull(test);
         Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
         Assert.assertEquals(OrderState.FAILED, test.getOrderState());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
     }
 
-    /**
-     * Test if a fulfilled order of a failed localidentity compute instance is definitely changed to
-     * failed.
-     *
-     * @throws InterruptedException
-     */
+    // test case: When running thread in the FulfilledProcessor and the InstanceState is Failed,
+    // the processFulfilledOrder() method must change the OrderState to Failed by adding in that
+    // list, and removed from the Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
     public void testRunProcessLocalComputeOrderInstanceFailed() throws Exception {
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
         orderInstance.setState(InstanceState.FAILED);
         order.setInstanceId(FAKE_INSTANCE_ID);
 
         mockCloudConnectorFactory(orderInstance);
 
-        Assert.assertNull(this.failedOrderList.getNext());
-
+        // exercise
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
+        this.thread.sleep(DEFAULT_SLEEP_TIME);
 
-        Thread.sleep(MIN_SLEEP_TIME);
-
-        Assert.assertNull(this.fulfilledOrderList.getNext());
-
+        // verify
         Order test = this.failedOrderList.getNext();
         Assert.assertNotNull(test);
         Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
         Assert.assertEquals(OrderState.FAILED, test.getOrderState());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
     }
 
+    // test case: When running thread in the FulfilledProcessor with OrderState Null must throw a
+    // ThrowableException.
+    @SuppressWarnings("static-access")
     @Test
-    public void testRunThrowableExceptionWhileTryingToProcessOrder()
+    public void testRunThrowableExceptionWhileTryingToProcessOrderStateNull()
             throws InterruptedException, UnexpectedException {
-        
+
+        // set up
         Order order = Mockito.mock(Order.class);
```

Comment:
Do we really need to mock this object?

Reply:
In that case, I think so, because Order is an abstract class